### PR TITLE
Add new APIv1 update response endpoint

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR adds a new endpoint to update a previously created record response:
* `PUT` `/api/v1/responses/{response_id}`

Notes:
* Admin users can update any response (even if that admin user didn't create the response in first place).
* Annotators can only update a response if the response was created by them.

### CODE REVIEW

File: src/argilla/server/schemas/v1/responses.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from uuid import UUID
+
+from pydantic import BaseModel
+
+
+class Response(BaseModel):
+    id: UUID
+    values: Dict[str, Any]
+    inserted_at: datetime
+    updated_at: datetime
```
Comment (frascuchon) [inline src/argilla/server/schemas/v1/responses.py:28]: Aligning the entity definition, I think we should add the user_id in the schema.

Comment (frascuchon) [reply src/argilla/server/schemas/v1/responses.py:28]: And the `record_id` for these endpoints.

Comment (jfcalvo) [reply src/argilla/server/schemas/v1/responses.py:28]: Yes, it makes sense.

Comment (jfcalvo) [reply src/argilla/server/schemas/v1/responses.py:28]: Done.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**